### PR TITLE
Restore docs from older releases

### PR DESCRIPTION
**Pull request by bf4**
The 0.8/0.9/0.10 releases all have different roots, so navigating the history
requires navigating each branch.

This PR pulls together some of that history for a better understanding of the library.

- [ ] review other linked sources

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -210,3 +212,5 @@ To run a single test suite
 Which can be further narrowed down to one test:
 
 `$ rake test TEST=path/to/test.rb TESTOPTS="--name=test_something"`
+
+:heart: :sparkling_heart: :heart:
```
Comment (joaomdmoura) [inline CONTRIBUTING.md]: :heart: :heart: :heart: :heart: :heart:

Comment (bf4) [inline CONTRIBUTING.md]: copied from the original contributing :)

**Comment by joaomdmoura** [comment]
Most of the content was already written, and have it all together seems nice!
LGTM :+1:

**Comment by bf4** [comment]
I have a few sources I wanted to review, also:

- https://groups.google.com/forum/#!searchin/rubyonrails-core/serializable/rubyonrails-core/mtHZiT67Ebo/3IyvwC9e1q4J
- https://groups.google.com/forum/#!searchin/rubyonrails-core/serializable/rubyonrails-core/GoWH8B2yi3o/bk7sbUq2ScwJ
- https://groups.google.com/forum/#!searchin/rubyonrails-core/serializable/rubyonrails-core/1SdvbsxjWc8/UGdxkd3_bWAJ
- https://groups.google.com/forum/#!searchin/rubyonrails-core/serializable/rubyonrails-core/MmW5lUVBUYA/o97VbkxSAogJ
- https://groups.google.com/forum/#!searchin/rubyonrails-core/serializable/rubyonrails-core/K8t4-DZ_DkQ/FukWpChXzKMJ
- https://groups.google.com/forum/#!topic/rails-api-core/8zu1xjIOTAM
- https://groups.google.com/forum/#!topic/rails-api-core/RkOHANFR4RM
- https://groups.google.com/forum/#!searchin/rails-api-core/serializable/rails-api-core/FRy7GMbGEMI/NBMHgdHnGlEJ
- https://groups.google.com/d/topic/rails-api-core/pGSWLiWUsCw/discussion
- https://groups.google.com/forum/#!searchin/rails-api-core/serializable/rails-api-core/YiFboKjGpGw/3CpjcgSd0XIJ
- https://groups.google.com/forum/#!searchin/rails-api-core/serializable/rails-api-core/pCRf4N9hwBM/LfWKSUMwRz8J
- https://groups.google.com/d/topic/rails-api-core/LoJap3IzOig/discussion
- https://groups.google.com/d/topic/rails-api-core/nym3xFTK0Zg/discussion
- https://groups.google.com/forum/#!topic/rails-api-core/FRy7GMbGEMI
- https://groups.google.com/forum/#!topic/rails-api-core/kxP0WxWzFEg

And Rails commits

```plain

5b2eb64ceb08cd005dc06b721935de5853971473
commit 5b2eb64ceb08cd005dc06b721935de5853971473
Author: José Valim <jose.valim@gmail.com>
Date:   Wed Nov 30 18:38:28 2011 +0100

    Revert "Implement ArraySerializer and move old serialization API to a new namespace."

    This reverts commit 8896b4fdc8a543157cdf4dfc378607ebf6c10ab0.

    Conflicts:

        activemodel/lib/active_model.rb
        activemodel/lib/active_model/serializable.rb
        activemodel/lib/active_model/serializer.rb
        activemodel/test/cases/serializer_test.rb

diff --git a/activemodel/CHANGELOG.md b/activemodel/CHANGELOG.md
index 95b682a..bd9ed99 100644
--- a/activemodel/CHANGELOG.md
+++ b/activemodel/CHANGELOG.md
@@ -5,14 +5,6 @@

     *Jon Leighton*

-*   Renamed (with a deprecation the following constants):
-
-        ActiveModel::Serialization     => ActiveModel::Serializable
-        ActiveModel::Serializers::JSON => ActiveModel::Serializable::JSON
-        ActiveModel::Serializers::Xml  => ActiveModel::Serializable::XML
-
-    *José Valim*

commit b6916e0b925d4c8b4487d574fe07b11440f2ec5e
Author: Xavier Noria <fxn@hashref.com>
Date:   Fri Nov 25 15:06:30 2011 -0800

    removes a Serialization constant left in the previous revert
    
commit 0a4035b12a6c59253cb60f9e3456513c6a6a9d33
Author: José Valim <jose.valim@gmail.com>
Date:   Fri Nov 25 19:29:39 2011 +0000

    Revert the serializers API as other alternatives are now also under discussion
commit fcacc6986ab60f1fb2e423a73bf47c7abd7b191d
Merge: 4565c87 696d01f
Author: José Valim <jose.valim@gmail.com>
Date:   Fri Nov 25 09:58:18 2011 +0000

    Merge branch 'serializers'

    This implements the ActiveModel::Serializer object. Includes code, tests, generators and guides.

    From José and Yehuda with love.

    Conflicts:
        railties/CHANGELOG.md
        
commit 696d01f7f4a8ed787924a41cce6df836cd73c46f
Author: José Valim <jose.valim@gmail.com>
Date:   Fri Nov 25 09:49:54 2011 +0000

    Add docs to serializers. Update CHANGELOGs.

commit 6d9f9b3b05f6ea55aad9853888912bcca4d81c9f
Author: José Valim <jose.valim@gmail.com>
Date:   Fri Nov 25 09:17:32 2011 +0000

    Add a hook for serializers in the scaffold generator (off for now).
commit ebb8ea22f30dc029f4a868a550828443b160f639
Author: José Valim <jose.valim@gmail.com>
Date:   Fri Nov 25 09:13:19 2011 +0000

    Add generators for serializers.
    
commit 28bcda4098985ae5eb2d18c5214d95cf04f3ea93
Author: José Valim <jose.valim@gmail.com>
Date:   Wed Nov 23 23:53:20 2011 +0000

    Rename UserSerializer to DefaultUserSerializer in tests.
    
commit 7fcc8c0a1f38c77b12cb6ffe81fb2887e6c60b85
Author: José Valim <jose.valim@gmail.com>
Date:   Wed Nov 23 23:45:27 2011 +0000

    Rely solely on active_model_serializer and remove the fancy constant lookup.

 6da52c617e2a075b9d9e7142786f316d8f2c7930
Author: José Valim <jose.valim@gmail.com>
Date:   Wed Nov 23 23:22:46 2011 +0000

    Remove listings from the serialization guide that won't make 3.2.
    
commit 8896b4fdc8a543157cdf4dfc378607ebf6c10ab0
Author: José Valim <jose.valim@gmail.com>
Date:   Wed Nov 23 23:18:13 2011 +0000

    Implement ArraySerializer and move old serialization API to a new namespace.

    The following constants were renamed:

      ActiveModel::Serialization     => ActiveModel::Serializable
      ActiveModel::Serializers::JSON => ActiveModel::Serializable::JSON
      ActiveModel::Serializers::Xml  => ActiveModel::Serializable::XML

    The main motivation for such a change is that `ActiveModel::Serializers::JSON`
    was not actually a serializer, but a module that when included allows the target to be serializable to JSON.

    With such changes, we were able to clean up the namespace to add true serializers as the ArraySerializer.
    
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -241,3 +278,94 @@ Features:
...
+
+## First Commit as [Rails Serializers 0.0.1](https://github.com/rails-api/active_model_serializers/commit/d72b66d4c5355b0ff0a75a04895fcc4ea5b0c65e)
+  (December 1, 2011).
+
+## Prehistory
+
+- [Changing Serialization/Serializers namespace to `Serializable` (November 30, 2011)](https://github.com/rails/rails/commit/8896b4fdc8a543157cdf4dfc378607ebf6c10ab0)
+  - [Merge branch 'serializers'. This implements the ActiveModel::Serializer object. Includes code, tests, generators and guides. From José and Yehuda with love.](https://github.com/rails/rails/commit/fcacc6986ab60f1fb2e423a73bf47c7abd7b191d)
+  - But [was reverted](https://github.com/rails/rails/commit/5b2eb64ceb08cd005dc06b721935de5853971473).
+    '[Revert the serializers API as other alternatives are now also under discussio(https://github.com/rails/rails/commit/0a4035b12a6c59253cb60f9e3456513c6a6a9d33)'.
```
Comment (bf4) [inline CHANGELOG.md]: @joaomdmoura Just updated this. I didn't realize the serializable was actually merged into Rails at one point. @josevalim @wycats what happened to `ActiveModel::Serializable`, `ActiveModel::Serializable::JSON`?

File: CHANGELOG.md [hunk 3]
```diff
@@ -241,3 +278,94 @@ Features:
...
+
+* First tagged version
+* Changes generators to always generate an ApplicationSerializer
+
+## [0.1.0 (December 21, 2011)](https://github.com/rails-api/active_model_serializers/commit/1e0c9ef93b96c640381575dcd30be07ac946818b)
+
+## First Commit as [Rails Serializers 0.0.1](https://github.com/rails-api/active_model_serializers/commit/d72b66d4c5355b0ff0a75a04895fcc4ea5b0c65e)
+  (December 1, 2011).
+
+## Prehistory
```
Comment (bf4) [inline CHANGELOG.md]: whoops, lost a [comment]( when I fixed a spelling error

```plain
@joaomdmoura Just updated this. I didn't realize the serializable was actually merged into Rails at one point. @josevalim @wycats what happened to ActiveModel::Serializable, ActiveModel::Serializable::JSON?
```

**Pull request merged**